### PR TITLE
DOC/BF: update a bunch more help text

### LINKED ISSUE #456: Document new configs in onyo-config's help

### LINKED ISSUE #569: install instructions insufficient
When I try to install Onyo like described in the README, it tells me it needs python. 
My installed version is Python3.11.2. The next step is activating the venv. This one fails for me with `zsh: command not found: python
`. 

Then I tried to follow the link `installation instructions`:
I shall need python, tig and a few others. Those are installed. Next step is the pip install. beeep, error: `zsh: command not found: pip
`

Of course I know how to install it then, but I think the instructions should be more clear for newbies.

### PR DESCRIPTION
**Pull request by aqw**
This PR is almost entirely improvements to the contents and formatting of help text. Lots was simply improved, and quite a few wrongs were righted. :-)

- `config`
- `edit`
- `history`
- `init`
- `shell-completion`
- `tree`

This PR also:
- removes the tab completion for the recently removed match options for `set` and `unset` (#557)
- fixes #569 
- fixes #456

### CODE REVIEW

File: onyo/commands/config.py [hunk 1]
```diff
@@ -16,15 +16,17 @@
...
 
 
 def config(args: argparse.Namespace) -> None:
     """
-    Set, query, and unset Onyo repository configuration options. These options
-    are stored in ``.onyo/config``. This file is tracked by git and are shared
-    with all other consumers of an Onyo repository.
+    Set, query, and unset Onyo repository configuration options.
+
+    These options are stored in ``.onyo/config``, which is is tracked by git and
```
Comment (bpoldrack) [inline onyo/commands/config.py:28]:
```suggestion
    These options are stored in ``.onyo/config``, which is tracked by git and
```

**Review by bpoldrack — APPROVED** [review]
Thanks much!

Spotted a word repetition, otherwise LGTM.

File: onyo/commands/config.py (hunk 1 above)
Comment (aqw) [reply onyo/commands/config.py]: Thanks for catching this. Fixed and pushed. :-)

**Comment by aqw** [comment]
I will wait for the tests to pass again and then merge. Thanks for reviewing this!

I don't know if I will get to all of the commands before Monday, but I do hope to get a few more done today. We'll see. :-)

**Pull request merged**